apps/settings/js/panels/firefox_sync/firefox_sync_panel.js

```diff
@@ -0,0 +1,288 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* global ERROR_UNVERIFIED_ACCOUNT */
+/* global LazyLoader */
+
+define(function(require) {
+  'use strict';
+
+  var SettingsListener = require('shared/settings_listener');
+  var SyncManagerBridge = require('panels/firefox_sync/manager_bridge');
+
+  const LOGGED_OUT_SCREEN = 'loggedout';
+  const LOGGED_IN_SCREEN = 'loggedin';
+
+  const ELEMENTS = [{
+    screen: LOGGED_OUT_SCREEN,
+    selector: '.fxsync-login',
+    event: 'click',
+    listener: 'enable'
+  }, {
+    screen: LOGGED_IN_SCREEN,
+    selector: '.fxsync-logout',
+    event: 'click',
+    listener: 'disable'
+  }, {
+    screen: LOGGED_IN_SCREEN,
+    selector: '.fxsync-user'
+  }, {
+    screen: LOGGED_IN_SCREEN,
+    selector: '.fxsync-sync-now',
+    event: 'click',
+    listener: 'sync'
+  }, {
+    screen: LOGGED_IN_SCREEN,
+    selector: '.fxsync-tos',
+    event: 'click',
+    listener: 'openTos'
+  }, {
+    screen: LOGGED_IN_SCREEN,
+    selector: '.fxsync-privacy',
+    event: 'click',
+    listener: 'openPrivacy'
+  }, {
+    screen: LOGGED_IN_SCREEN,
+    selector: '.fxsync-collections-history'
+  }, {
+    screen: LOGGED_IN_SCREEN,
+    selector: '.fxsync-collections-passwords'
+  }, {
+    screen: LOGGED_IN_SCREEN,
+    selector: '.fxsync-last-sync'
+  }, {
+    screen: LOGGED_IN_SCREEN,
+    selector: '.fxsync-unverified'
+  }];
+
+  function getElementName(str) {
+    str = str.toLowerCase().replace('.fxsync-', '');
+    return str.replace(/-([a-z])/g, g => {
+      return g[1].toUpperCase();
+    });
+  }
+
+  function FirefoxSyncPanel(panel) {
+    this.panel = panel;
+    this.elements = {};
+    this.screens = {
```

Comment:
Please move major querySelector sentences into panel.js, and refer them from `this._elements` here.